### PR TITLE
Wireup Express Generator

### PR DESCRIPTION
**Pull request by beverts312**
Here are features included:  
* Add express package & typing to package.json
* Enhance docker related files  
* Add Express app starter file
* Wireup generator  

In order to make this work I had to make the following modifications to the boilerplate:  
* add tslint gulp task - Alternatively we could remove the tslint dependencies  
* remove package from build step - I dont think this should be required as part of the build

### CODE REVIEW

**Comment by beverts312** [comment]
@calebcartwright  All of these checks are great for my self esteem

File: generators/app/templates/express-api/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
-FROM alpine:3.3
```
Comment (calebcartwright) [inline generators/app/templates/express-api/Dockerfile]: @beverts312 is this better than the official node alpine image? https://hub.docker.com/_/node/
FROM node:7-apline

Comment (beverts312) [inline generators/app/templates/express-api/Dockerfile]: I wouldn't say that it is better necessarily, just you know exactly what you are getting

Comment (calebcartwright) [inline generators/app/templates/express-api/Dockerfile]: In terms of knowing what all is inside the image, knowing version of node, or both? I wonder if there is any size difference. Bet node one is bigger

Comment (beverts312) [inline generators/app/templates/express-api/Dockerfile]: @calebcartwright 
of knowing what other things they add in there, obviously its not that hard as you can hop over to the node image github page and check it out.
We could do that if you want, I typically dont bother with it

Comment (beverts312) [inline generators/app/templates/express-api/Dockerfile]: some people thing its better practice, I dont think its a big deal. But if it makes openshift work ;)

**Pull request merged**